### PR TITLE
Updates Solgov/Sifguard uniforms

### PR DESCRIPTION
**Pull request by Novacat**
Since BYOND increased the maximum of bitwise operations, it is possible to add a few new slots. I decided to port Bay's Rank and Dept slots, as well as remove the now defunct Torso slot.

Also ports some missing uniform/suit items from Baystation, and fixes a bug with the Marine rank sprite

Also updated the uniform, suit, hats, and accessories to relatively latest baycode, when I noticed our uniforms looked kinda bad I realized that a lot of stuff was not ported.

### CODE REVIEW

File: code/modules/clothing/under/solgov.dm [hunk 1]
```diff
@@ -313,3 +468,33 @@
...
+
+/obj/item/clothing/under/terran/navy/utility
+	name = "ICCGN utility uniform"
+	desc = "A comfortable black utility jumpsuit. Worn by the ICCG Navy."
+	icon_state = "terranutility"
+	worn_state = "terranutility"
+
+/obj/item/clothing/under/terran/navy/service
+	name = "ICCGN service uniform"
+	desc = "The service uniform of the ICCG Navy, for low-ranking crew."
```
Comment (Atermonera) [inline code/modules/clothing/under/solgov.dm:490]: army

File: code/modules/clothing/under/solgov.dm [hunk 2]
```diff
@@ -313,3 +468,33 @@
...
+	icon_state = "terranservice"
+	worn_state = "terranservice"
+	armor = list(melee = 5, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 5, rad = 0)
+	siemens_coefficient = 0.9
+
+/obj/item/clothing/under/terran/navy/service/command
+	name = "ICCGN command service uniform"
+	desc = "The service uniform of the ICCG Navy, for high-ranking crew."
```
Comment (Atermonera) [inline code/modules/clothing/under/solgov.dm:498]: army

File: code/modules/power/smes.dm [hunk 3]
```diff
@@ -154,7 +154,7 @@
 
 	//inputting
 	if(input_attempt && (!input_pulsed && !input_cut) && !grid_check)
-		target_load = clamp((capacity-charge)/SMESRATE, 0, input_level)	// Amount we will request from the powernet.
+		target_load = CLAMP((capacity-charge)/SMESRATE, 0, input_level)	// Amount we will request from the powernet.
```
Comment (Atermonera) [inline code/modules/power/smes.dm:157]: Pretty sure I merged that fix?

File: polaris.dme [hunk 4]
```diff
@@ -1416,8 +1416,8 @@
 #include "code\modules\ai\ai_holder_targeting.dm"
 #include "code\modules\ai\interfaces.dm"
 #include "code\modules\ai\say_list.dm"
-#include "code\modules\ai\aI_holder_subtypes\simple_mob_ai.dm"
-#include "code\modules\ai\aI_holder_subtypes\slime_xenobio_ai.dm"
+#include "code\modules\ai\ai_holder_subtypes\simple_mob_ai.dm"
+#include "code\modules\ai\ai_holder_subtypes\slime_xenobio_ai.dm"
```
Comment (Atermonera) [inline polaris.dme:1420]: ???

**Comment by Cerebulon** [comment]
In a broader sense, copy-pasting lore stuff from Bay isn't going to work because our settings are extremely incompatible in many regards and even rewriting most of these would be introducing uniforms and by extension concepts for about a dozen granular service branches we don't particularly have a want or need for (We aren't a military-centric setting like Bay is).
I think it might be best to just port the updated sprites/refactors for stuff we already did have. And berets I guess are pretty much harmless.
Also yeah, there's no army.

**Comment by Novacat** [comment]
I am not too intimately familiar on Polaris lore, so I mostly just copy-paste the Bay names and descriptions and I figure they would be changed again once a purpose or role for that item is determined here. Will do the requested changes, but I will keep the army pathing and iconstates for the sake of compatibility.

File: code/modules/clothing/under/solgov.dm [hunk 5]
```diff
@@ -313,3 +468,33 @@
...
+	name = "master ICCGN uniform"
+	desc = "You shouldn't be seeing this."
+	icon = 'icons/obj/clothing/uniforms_solgov.dmi'
+	item_icons = list(slot_w_uniform_str = 'icons/mob/uniform_solgov.dmi')
+	armor = list(melee = 5, bullet = 0, laser = 5, energy = 5, bomb = 0, bio = 5, rad = 5)
+	siemens_coefficient = 0.8
+
+/obj/item/clothing/under/terran/navy/utility
+	name = "ICCGN utility uniform"
+	desc = "A comfortable black utility jumpsuit. Worn by the ICCG Navy."
```
[the remark this answers is not in the document]
Comment (Novacat) [reply code/modules/clothing/under/solgov.dm:484]: ICCGN is not Solgov, they are basically Bay's equivalent to faction-level Antagonists for Solgov. They started off as Space Soviets but Bay has been trying to move away from that.

File: code/modules/power/smes.dm (hunk 3 above)
Comment (Novacat) [reply code/modules/power/smes.dm:157]: I had to put that fix for me since I still use 512.

File: polaris.dme (hunk 4 above)
Comment (Novacat) [reply polaris.dme:1420]: Dreammaker being silly.

File: code/modules/clothing/head/solgov.dm [hunk 6]
```diff
@@ -10,7 +10,12 @@
 		slot_r_hand_str = "lightbluesoft",
 		)
 
-/obj/item/clothing/head/soft/sol/expedition
+/obj/item/clothing/head/soft/solgov/veteranhat
+	name = "veteran hat"
+	desc = "It's a tacky black ballcap bearing the yellow service ribbon of the Gaia Conflict."
```
[the remark this answers is not in the document]
Comment (Novacat) [reply code/modules/clothing/head/solgov.dm:15]: I am leaving new stuff with the default names/descs so you guys can figure out what you want to do with it (if anything).

File: code/modules/clothing/head/solgov.dm [hunk 7]
```diff
@@ -113,124 +143,248 @@
...
-/obj/item/clothing/head/beret/sol/health
+/obj/item/clothing/head/beret/solgov/health
 	name = "health service beret"
 	desc = "A white beret denoting service in the Interstellar Health Service. For medics that are more inclined towards style than safety."
 	icon_state = "beret_white"
 
-/obj/item/clothing/head/beret/sol/expedition
+/obj/item/clothing/head/beret/solgov/marcom
+	name = "\improper MARSCOM beret"
+	desc = "A red beret with a gold insignia, denoting service in the SCGDF Mars Central Command. For brass who are more inclined towards style than safety."
```
[the remark this answers is not in the document]
Comment (Novacat) [reply code/modules/clothing/head/solgov.dm:210]: New item

File: code/modules/clothing/head/solgov.dm [hunk 8]
```diff
@@ -113,124 +143,248 @@
...
+	icon_state = "beret_redgold"
+
+/obj/item/clothing/head/beret/solgov/stratcom
+	name = "\improper STRATCOM beret"
+	desc = "A grey beret with a silver insignia, denoting service in the SCGDF Strategic Command. For intelligence personnel who are more inclined towards style than safety."
```
[the remark this answers is not in the document]
Comment (Novacat) [reply code/modules/clothing/head/solgov.dm:215]: New item

File: code/__defines/items_clothing.dm [hunk 9]
```diff
@@ -17,28 +17,28 @@
...
+#define ACCESSORY_SLOT_UTILITY  0x1
+#define ACCESSORY_SLOT_WEAPON   0x2
 #define ACCESSORY_SLOT_ARMBAND  0x4
-#define ACCESSORY_SLOT_DECOR    0x8
-#define ACCESSORY_SLOT_MEDAL    0x20
-#define ACCESSORY_SLOT_TIE		0x40
-#define ACCESSORY_SLOT_INSIGNIA 0x80
-#define ACCESSORY_SLOT_OVER		0x100
+#define ACCESSORY_SLOT_RANK     0x8
+#define ACCESSORY_SLOT_DEPT     0x20
```
[the remark this answers is not in the document]
Comment (Novacat) [reply code/__defines/items_clothing.dm:29]: This is mainly because ranks/department insignia are intended to have a limit on how many you can attach.

**Pull request merged**